### PR TITLE
chore(docs): Migrate developer specific docs from wiki.trezor.io to d…

### PR DESCRIPTION
**Pull request by sime**
…ocs.trezor.io

The PR is getting too big, so will be splitting it up.

Related but doesn't completely close [#2304](https://github.com/trezor/trezor-firmware/issues/2304)

Majority of the content is copy and pasted, with some obvious edits.

If you spot obvious errors, please offer a suggestion and I will almost blindly accept them.

### CODE REVIEW

**Comment by sime** [comment]
I need a pointer why this fixup failed.

**Comment by matejcik** [comment]
all of `connect` should go to the Suite repo

`trezorctl` seems kinda superfluous given that everything there can be obtained by judicious use of `trezorctl <x> --help`

we have very nice docs about the signing flow at `common/communication`. The `message-workflows` file should probably go there too -- but a lot of it is outdated.

Probably all the new files should have a header like "Migrated from old Trezor Wiki, may be outdated"

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - [Legacy](legacy/index.md)
 - [Python](python/index.md)
   - [trezorlib](python/trezorlib.md)
+  - [Trezor CTL](python/trezorctl.md)
```
Comment (matejcik) [inline docs/SUMMARY.md:21]: this should be `trezorctl`, nobody calls it "Trezor CTL"

File: docs/developers/android.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+app. Note that the mobile phone needs to support OTG connection and the
+user needs to have an OTG cable to connect Trezor device to the
+phone.
+
+*See more documentation on the [GitHub page][].*
+
+Examples of Android applications that integrate with Trezor using the
+trezor-android API:
+
+-   [Mycelium][]
```
Comment (matejcik) [inline docs/developers/android.md:13]: i see this syntax `[something][]` throughout, is it intentional? AFAIK you can just remove the trailing `[]`

**Comment by matejcik** [comment]
> I need a pointer why this fixup failed.

the "block-fixup" job does what it says on the tin: blocks any PRs that have `fixup!` commits.
this is expected to fail for as long as the review cycle is running. afterwards you autosquash the fixups to clear the check.

File: docs/developers/android.md (hunk 2 above)
Comment (sime) [reply docs/developers/android.md:13]: I used pandoc.org/try/ on the smaller pages, and this is a relic of it. The links worked, so I left it.

I'll do some clearing.

File: docs/developers/android.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Android
```
[the remark this answers is not in the document]
Comment (sime) [reply docs/developers/android.md:1]: Good point. Will get old wiki to redirect to the Android repo.

[the remark this answers is not in the document]
Comment (sime) [reply docs/developers/android.md]: Android page is removed.

File: docs/misc/contributing.md [hunk 4]
```diff
@@ -15,3 +15,42 @@ Your Pull Request should follow these criteria:
...
+
+-   **Bitcoin clones** should be added to the [common/defs/bitcoin][]
+    subdirectory as separate .json files
+-   **Ethereum networks/chains** should be added to the
+    [ethereum-lists/chains][]
+-   **Ethereum tokens** should be added to the [ethereum-lists/tokens][]
+    repository
+-   **NEM mosaics** should be added to the
+    [common/defs/nem/nem_mosaics.json][] file
+-   **other coins** should be added to the
```
[the remark this answers is not in the document]
Comment (sime) [reply docs/misc/contributing.md]: Cleared

**Comment by matejcik** [comment]
> I believe you are thinking that way because it comes from the Suite repo? Considered that myself, though, Connect is more closely related to Firmware. Because it enables developers to interact with the firmware, rather than anything to do with Suite.
> 

It does not make sense to maintain Connect docs in the firmware repo; they need to be maintained in the Connect repo (which is currently in `trezor-suite`). Connect presents its own API that follows the firmware API but is separate -- we firmware devs actually know very little about the Connect API and how to use it.

**Comment by sime** [comment]
> It does not make sense to maintain Connect docs in the firmware repo; they need to be maintained in the Connect repo (which is currently in `trezor-suite`). Connect presents its own API that follows the firmware API but is separate -- we firmware devs actually know very little about the Connect API and how to use it.

Clear to me now. I removed Trezor Connect content.

File: docs/common/message-workflows.md [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+## SignTx
+
+Signing a transaction is a little bit complicated. The reason is that
+transactions can be several hundred kilobytes in size, but Trezor has
+only 64 kilobytes memory. So it is the task of the computer to split the
+transactions in small pieces and send only those pieces that Trezor
+requested. The general workflow is given below
+
+![](Developers_guide_signtx_workflow.png)
```
Comment (matejcik) [inline docs/common/message-workflows.md:87]: this file should be also moved to `common`

File: docs/common/message-workflows.md [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+## GetPublicKey
+
+The message GetPublicKey can be used to get a bip-32 master public key
+from the trezor or to display it to the user. The field address_n gives
+the bip-32 path to the master key. The field ecdsa_curve_name can be
+used to get Ed25519 or
+NIST256P1 public keys.
+
+## SignTx
```
Comment (matejcik) [inline docs/common/message-workflows.md:79]: please add a header pointing to `communication/bitcoin-signing.md`, saying that the below might be imprecise / obsolete

File: docs/common/message-workflows.md [hunk 7]
```diff
@@ -0,0 +1,314 @@
+# Message Workflows
```
Comment (matejcik) [inline docs/common/message-workflows.md:1]: please also add "this page was migrated" here

Comment (sime) [reply docs/common/message-workflows.md:1]: Resolved in 0268958.

File: docs/common/message-workflows.md (hunk 6 above)
Comment (sime) [reply docs/common/message-workflows.md:82]: Resolved in 0268958.

File: docs/common/message-workflows.md (hunk 5 above)
Comment (sime) [reply docs/common/message-workflows.md:94]: Resolved in 0268958.

**Review by matejcik — APPROVED** [review]

**Comment by matejcik** [comment]
please squash the fixups and force-push, so that we can merge

**Pull request merged**